### ISSUE TITLE
Support adding default values to generated schema

### ISSUE THREAD

**Issue by kingwill27** [ch-robinson/dotnet-avro#57]
@dstelljes I'm creating this issue based on your comment on #7 

Since default values aren't accessible via reflection, maybe an easier way to implement this functionality would be via an annotation? Something to the effect of

```
public class Message
{
  [AvroDefaultValue(null)]
  public int? Property { get; set; } = null;
}
```

**Comment by dstelljes** [comment]
So these are probably the prerequisites:

1. Add default value properties to `Schema` and `TypeResolution` (#7)
2. Support reading/writing default values from JSON (#2)

From there, we could consider shipping annotations, or you could write your own and extend the resolver.

**Comment by digitalpacman** [comment]
Since this generator doesn't support default values, it's impossible to use it for backward compatible types.  Avro requires default values be specified. Allowing null isn't enough.

**Comment by dstelljes** [comment]
@digitalpacman Say the schema builder (assuming it’s unable to get a default value from an attribute or whatever) set a known-compatible default value, like 0 for an `"int"` schema. When you say that allowing null isn’t enough, is that what you’re getting at?

To your comment on #7 suggesting `[DefaultValue]`, that’d definitely be preferable to shipping our own annotation (and it might be good to support some other features of System.ComponentModel as well). Either way, the comment about prereqs stands; to properly support reading/writing default values from/to JSON, we have to complete #2.

**Comment by dstelljes** [comment]
In 8.0.0, we’re adding support for the `[DefaultValue]` attribute from System.ComponentModel. As the original request is satisfied, I’ll close this issue once the functionality is released.

Since there are limitations to what can be defined by attribute (and the point about coupling serialization-specific attributes to other infrastructure is well taken), we’d be open to exploring something like the fluent interface above, but it’s out of scope for the next release. Proposals are welcome as [discussions](https://github.com/ch-robinson/dotnet-avro/discussions).

**Comment by dstelljes** [comment]
Support for the `[DefaultValue]` attribute is released with 8.0.0-pre.1.
